sdk/textanalytics/out/azure/ai/textanalytics/_models.pyi

```diff
@@ -105,6 +103,7 @@ class DocumentError(DictMixin):
     error: Any = ...
     is_error: bool = ...
     def __init__(self, **kwargs: Any) -> None: ...
+    def __getattr__(self, attr: Any) -> None: ...
```

Comment:
How're we using the getattr here?

Reply:
For batching, each method returns a heterogeneous list of result and error objects in the order that the user passed in the input documents. If a user is iterating through the response and tries to access a result attribute that doesn't exist on an error object, this will raise an AttributeError, but with a custom message that describes the input document ID and error message, e.g.

`AttributeError: 'DocumentError' object has no attribute 'sentiment'. The service was unable to process this document:
Document Id: 0
Error: invalidDocument - Document text is empty.`

Reply:
Given that there's a one-to-one mapping between input and output - do you think there would be any benefit to returning a dictionary of Input->Result?

Reply:
This is a really good point. We've considered it after watching people struggle in the ux studies with taking a result and wanting to use a subset of the input in another method. I'll bring it up :)